src/plugins/typingIndicator.tsx

```diff
@@ -0,0 +1,267 @@
+
+interface ITyping {
+    channelId: string;
+    userId: string;
+}
+
+interface IMessageCreate {
+    type: "MESSAGE_CREATE";
+    optimistic: boolean;
+    isPushNotification: boolean;
+    channelId: string;
+    message: Message;
+}
+
+const dotsIcon = () => (
+    <svg
+        className="circleIcon"
+        height="24"
+        width="24"
+        viewBox="0 0 24 24"
+        fill="currentColor"
+    >
+        <path
+            d="M12,1.5A1.5,1.5,0,1,1,13.5,3,1.5,1.5,0,0,1,12,1.5Zm-6,0A1.5,1.5,0,1,1,7.5,3,1.5,1.5,0,0,1,6,1.5Zm-6,0A1.5,1.5,0,1,1,1.5,3,1.5,1.5,0,0,1,0,1.5Z"
+            transform="translate(4.5 11)"
+        />
+    </svg>
+);
+
+const typingUsers: Array<ITyping> = [];
+let toolTipString: string = "";
+
+let buttonType: any;
+let buttonProps: any;
+let buttonChildren: any;
+
+function ContextMenuElement() {
+    const items: Array<JSX.Element> = typingUsers.map(u => {
+        const user = UserStore.getUser(u.userId);
+        if (!user) return <></>;
+
+        return <Menu.MenuItem
+            id={user.id}
+            key={user.id}
+            label={user.username}
+            action={() => {
+                openPrivateChannel(user.id);
+            }}
+        />;
+    });
+
+    return <Menu.ContextMenu
+        navId="typing-indicator-menu"
+        onClose={() => FluxDispatcher.dispatch({ type: "CONTEXT_MENU_CLOSE" })}>
+        {items}
+    </Menu.ContextMenu>;
+}
+
+let forceRenderIndicator: React.DispatchWithoutAction;
+
+const addUser = (channelId: string, userId: string) => {
+
+    const hasUser = typingUsers.some(o => o.channelId === channelId && o.userId === userId);
+
+    if (!hasUser) {
+        typingUsers.push({ channelId, userId });
+        UpdateElement();
+    }
+};
+
+const removeUser = (channelId: string, userId: string) => {
+
+    const userIndex = typingUsers.findIndex(o => {
+        return o.channelId === channelId && o.userId === userId;
+    });
+
+    if (userIndex !== -1) {
+        typingUsers.splice(userIndex, 1);
+        UpdateElement();
+    }
+};
+
+const UpdateElement = () => {
+
+    if (!buttonProps || !forceRenderIndicator)
+        return;
+
+    if (typingUsers.length === 0) {
+        toolTipString = Settings.plugins?.TypingIndicator?.alwaysShow ? Settings.plugins?.TypingIndicator?.emptyMessage : "";
+        forceRenderIndicator();
+        return;
+    }
+
+    if (typingUsers.length === 1) {
+        const user = UserStore.getUser(typingUsers.values().next().value.userId);
+        toolTipString = `${user.username} is typing...`;
+        forceRenderIndicator();
+        return;
+    }
+
+    toolTipString = "";
+    for (const user of typingUsers) {
+        if (user && user.userId) {
+            const member = UserStore.getUser(user.userId);
+            if (!member) continue;
+
+            toolTipString += `${member.username}, `;
+        }
+    }
+    toolTipString += "are typing...";
```

Comment:
```suggestion
    toolTipString = toolTipString.slice(0, -2) + "are typing...";
``` to remove the last comma